### DISCUSSION TITLE
Episode 203: Rotates All the Way Down with Sean Parent (Part 2)

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#102] [Episode]
Episode Link: https://adspthepodcast.com/2024/10/11/Episode-203.html

Feel free to write comments or ask questions 🙂

**Comment by VincentZalzal** [comment]
When you asked for other under-utilized algorithms, I was also shouting `nth_element` in my head!

Also, I had an algorithmic question while listening: since the most frequent use-case for rotate is the 1-rotate, would it be worth it to have a special implementation for that (in the same function or in a new one, whichever)? I feel like you could implement it with a single straightforward loop of N moves with one temporary element all along (and one final move). It is very memory-friendly, and if you implement swap as three moves, then it has almost 3 times fewer moves than the three reverses implementation, if I'm not mistaken.
